File: projects/engineering.purdue.edu/kak/scriptingwithobjects/swocode/chap4/MultiLine.py
#!/usr/bin/env python
  
## MultiLine.py 

import re

input_string = "abcd\nefgh\nijkl\n"                                 #(A)

# CASE 1:
regex = r'abcd'                                                     #(B)
m = re.search( regex, input_string )                                #(C)
if m: print "match successful"                 # match successful
else: print "no success"

# CASE 2:
regex = r'^abcd$'                                                   #(D)
m = re.search( regex, input_string )                                #(E)
if m: print "match successful"
else: print "no success"                       # no success

# CASE 3:
regex = r'^ab.*kl$'                                                 #(F)
m = re.search( regex, input_string )                                #(G)
if m: print "match successful"
else: print "no success"                       # no success

# CASE 4:
regex = r'^ab.*kl$'                                                 #(H)
m = re.search( regex, input_string, re.DOTALL )                     #(I)
if m: print "match successful"                 # match successful
else: print "no success"

# CASE 5:
regex = r'^abcd$'                                                   #(J)
m = re.search( regex, input_string, re.MULTILINE )                  #(K)
if m: print "match successful"                 # match successful
else: print "no success" 

# CASE 6:
regex = r'^ab.*kl$'                                                 #(L)
m = re.search( regex, input_string, re.MULTILINE )                  #(M)
if m: print "match successful"
else: print "no success"                       # no success

# CASE 7:
regex = r'^ab.*gh$'                                                 #(N)
m = re.search( regex, input_string, re.MULTILINE | re.DOTALL )      #(O)
if m: print "match successful"                 # match successful
else: print "no success"

# CASE 8:
regex = r'\Aab.*kl\Z'                                               #(P)
m = re.search( regex, input_string, re.MULTILINE | re.DOTALL )      #(Q)
if m: print "match successful"
else: print "no success"                       # no success

File: projects/engineering.purdue.edu/scriptingwithobjects/swocode/chap4/MultiLine.pl
#!/usr/bin/perl -w
  
## MultiLine.pl   

use strict;  

my $input_string = "abcd\nefgh\nijkl\n";                            #(A)

# CASE 1:
my $regex = 'abcd';                                                 #(B)
if ( $input_string =~ /$regex/ ) {                                  #(C)
    print "match successful\n";           # match successful        
} else {
    print "no success\n";
}

# CASE 2:
$regex = '^abcd$';                                                  #(D) 
if ( $input_string =~ /$regex/ ) {                                  #(E)
    print "match successful\n";           
} else {
    print "no success\n";                 # no success
}

# CASE 3:
$regex = '^ab.*kl$';                                                #(F)
if ( $input_string =~ /$regex/ ) {                                  #(G)
    print "match successful\n";           
} else {
    print "no success\n";                 # no success
}

# CASE 4:
$regex = '^ab.*kl$';                                                #(H)
if ( $input_string =~ /$regex/s ) {                                 #(I)
    print "match successful\n";           # match successful 
} else {
    print "no success\n";
}

# CASE 5:
$regex = '^abcd$';                                                  #(J)
if ( $input_string =~ /$regex/m ) {                                 #(K)
    print "match successful\n";           # match successful        
} else {
    print "no success\n";
}

# CASE 6:
$regex = '^ab.*kl$';                                                #(L)
if ( $input_string =~ /$regex/m ) {                                 #(M)
    print "match successful\n";           
} else {
    print "no success\n";                 #  no success
}

# CASE 7:
$regex = '^ab.*gh$';                                                #(N)
if ( $input_string =~ /$regex/sm ) {                                #(O)
    print "match successful\n";           # match successful  
} else {
    print "no success\n";
}

# CASE 8:
$regex = '\Aab.*kl\Z';                                              #(P)
if ( $input_string =~ /$regex/sm ) {                                #(Q)
    print "match successful\n";           # match successful 
} else {
    print "no success\n";
}
